src/main/protocol-handler/router.ts

```diff
@@ -0,0 +1,103 @@
+import logger from "../logger";
+import * as proto from "../../common/protocol-handler";
+import Url from "url-parse";
+import { LensExtension } from "../../extensions/lens-extension";
+import { broadcastMessage } from "../../common/ipc";
+
+export interface FallbackHandler {
+  (name: string): Promise<boolean>;
+}
+
+export class LensProtocolRouterMain extends proto.LensProtocolRouter {
+  private missingExtensionHandlers: FallbackHandler[] = [];
+
+  /**
+   * Find the most specific registered handler, if it exists, and invoke it.
+   *
+   * This will send an IPC message to the renderer router to do the same
+   * in the renderer.
+   */
+  public async route(rawUrl: string): Promise<void> {
+    try {
+      const url = new Url(rawUrl, true);
+
+      if (url.protocol.toLowerCase() !== "lens:") {
+        throw new proto.RoutingError(proto.RoutingErrorType.INVALID_PROTOCOL, url);
+      }
+
+      logger.info(`${proto.LensProtocolRouter.LoggingPrefix}: routing ${url.toString()}`);
+
+      switch (url.host) {
+        case "internal":
```

Comment:
Does this mean non-extension protocol URLs are like `lens://internal`? Maybe consider more "friendly" name because these links might also be published / used from outside of the app. One candidate is that we route everything except `extensions` to internal.

Reply:
do you mean ignoring the `host` of the URI if it is not `extension` or have some way to route based on `host` as well as `pathname`? Or could we change this to something like `app`? The non-extension protocol URIs are `lens://app`.

Reply:
I meant route based on `host` + `pathname` but if that's complicated then `lens://app` sounds also good.

Reply:
It is more complicated since `host` follow different rules from `pathname` and it doesn't gain us anything in terms of flexibility (imo).

Reply:
Ok, let's use `lens://app`.

Reply:
👍